--- whatsmeow/msgsecret.go
+++ whatsmeow.mod/msgsecret.go
@@ -230,7 +230,7 @@
 		options[i] = &waProto.PollCreationMessage_Option{OptionName: proto.String(option)}
 	}
 	return &waProto.Message{
-		PollCreationMessage: &waProto.PollCreationMessage{
+		PollCreationMessageV3: &waProto.PollCreationMessage{
 			Name:                   proto.String(name),
 			Options:                options,
 			SelectableOptionsCount: proto.Uint32(uint32(selectableOptionCount)),
